data-solutions/data-platform-foundations/resources/main.tf

```diff
@@ -163,39 +163,34 @@ module "bucket-transformation" {
 ###############################################################################
 #                                 Bigquery                                    #
 ###############################################################################
-module "bigquery-datasets-dwh" {
+
+module "bigquery-datasets-datamart" {
   source     = "../../../modules/bigquery-dataset"
-  project_id = module.project-dwh.project_id
+  project_id = module.project-datamart.project_id
 
-  for_each = var.dwh_bq_datasets
+  for_each = var.datamart_bq_datasets
   id       = each.value.id
   location = each.value.location
-
-  access = {
-    owner  = { role = "OWNER", type = "user" }
-    reader = { role = "READER", type = "user" }
-  }
-
+  access   = each.value.access
   access_identities = {
-    owner  = module.transformation-default-service-accounts.email
-    reader = module.dwh-default-service-accounts.email
+    owner = module.datamart-default-service-accounts.email
   }
+  #   access_identities = each.value.access_identities
 }
 
-module "bigquery-datasets-datamart" {
+module "bigquery-datasets-dwh" {
   source     = "../../../modules/bigquery-dataset"
-  project_id = module.project-datamart.project_id
+  project_id = module.project-dwh.project_id
 
-  for_each = var.datamart_bq_datasets
+  for_each = var.dwh_bq_datasets
   id       = each.value.id
   location = each.value.location
-
-  access = {
-    owner = { role = "OWNER", type = "user" }
-  }
+  access   = each.value.access
```

Comment:
We need to merge `access` and `access_identities` configured in variables with the one we need from the solution to work, a logic similar to the one implemented for pubSub permissions: https://github.com/terraform-google-modules/cloud-foundation-fabric/blob/d7825f9e915901d76ff0e35e6eb580629a7286bc/data-solutions/data-platform-foundations/resources/main.tf#L20-L30